src/main/java/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
+        int newQuantity = currentQuantity - quantity;
+        if (newQuantity < 0) {
```

Comment:
The check for newQuantity being negative is unnecessary because this condition will never be true due to the previous check (currentQuantity < quantity) that throws an exception. You can safely remove this check.